Replicator/Checkpointer.cc

```diff
@@ -234,9 +235,18 @@ namespace litecore { namespace repl {
         // Existing documents of are considered in the default collection
         // and they should keep the same docID or current checkpointers would be
         // inaccessible.
+        bool useSha1 = true;
         if (_collection != nullptr && !(_collection->getSpec() == kC4DefaultCollectionSpec)) {
-            enc.writeString(_collection->getSpec().name);
-            enc.writeString(_collection->getSpec().scope);
+            auto spec = _collection->getSpec();
+            auto index = _options->collectionSpecToIndex().at(spec);
+            enc.writeString(spec.name);
+            enc.writeString(spec.scope);
+
+            // CBL-501: Push only and pull only checkpoints create conflict
+            // So include them in the derivation
+            enc.writeBool(_options->pull(narrow_cast<unsigned int>(index)) != kC4Disabled);
```

Comment:
Why not to use CollectionIndex, in lieu of unsigned int?

Reply:
Reasonable point